src/js/background/context-menu.js

```diff
@@ -0,0 +1,371 @@
+    canUserGenerateAliases ? relayContextMenus.menus.create(staticMenuData.generateAliasEnabled) : relayContextMenus.menus.create(staticMenuData.generateAliasDisabled);
+
+    const userHasSomeAliasesCreated = (await relayContextMenus.utils.getUserStatus.getNumberOfAliases() > 0);
+    
+    const aliases = await relayContextMenus.utils.getAliases();
+
+    // Create Use Existing Alias submenu
+    if (currentWebsite &&  await relayContextMenus.utils.getGeneratedForHistory(currentWebsite) && userHasSomeAliasesCreated ) {
+      await relayContextMenus.menus.create(staticMenuData.existingAlias, {
+        createExistingAliases: true,
+        parentMenu: staticMenuData.useExistingAliasFromWebsite,
+        exisitingSite: true,
+        currentWebsite
+      }, aliases);
+    } 
+
+    // Create "Recent Aliases…" menu
+    if ( userHasSomeAliasesCreated ) {
+      await relayContextMenus.menus.create(staticMenuData.existingAlias, {
+        createExistingAliases: true,
+        parentMenu: staticMenuData.useExistingAlias,
+        exisitingSite: false,
+        currentWebsite
+      }, aliases)
+    }
+
+    // Create "Manage all aliases" link
+    await relayContextMenus.menus.create(staticMenuData.manageAliases);
+
+    // Generate upgrade menu item for non-premium users
+    const canUserUpgradeToPremium = await relayContextMenus.utils.getUserStatus.canUpgradeToPremium();
+    if (canUserUpgradeToPremium) {
+      await relayContextMenus.menus.create([staticMenuData.upgradeToPremiumSeperator, staticMenuData.upgradeToPremium]);
+    }
+
+    // Set listerners
+    browser.storage.onChanged.addListener(relayContextMenus.listeners.onLocalStorageChange);        
+
+    // Refresh menus
+    await browser.menus.refresh();
+
+    return Promise.resolve(1)
+
+  },
+  listeners: {
+    onFillInAddressWithAliasId: async (info, tab) => {
+      // Trim the context menu id to get the alias reference ID
+      const selectedAliasId = info.menuItemId.replace(
+        "fx-private-relay-use-existing-alias_",
+        ""
+      );
+
+      // Get stored alias data
+      const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+
+      // Select the correct alias from the stored alias data
+      const selectedAliasObject = relayAddresses.filter((alias) => {
+        return alias.id === parseInt(selectedAliasId, 10);
+      });
+
+      browser.tabs.sendMessage(
+        tab.id,
+        {
+          type: "fillTargetWithRelayAddress",
+          targetElementId: info.targetElementId,
+          relayAddress: selectedAliasObject[0],
+        },
+        {
+          frameId: info.frameId,
+        }
+      );
+      
+    },
+    onLocalStorageChange: async (changes, area) => {
+      let changedItems = Object.keys(changes);
+      for (let item of changedItems) {
+        if (item === "relayAddresses") {
+          await relayContextMenus.init();
```

Comment:
`init` calls `browser.storage.onChanged.addListener(relayContextMenus.listeners.
onLocalStorageChange);` again - is that safe?

Reply:
There wasn't any visual side effects of calling this too many times but it's a good call out regardless. I conditionally remove it if it exists each time init is called. 

Revised in 2437ed69f721c92089de045f8b7c1dee8c5e1fff